### PR TITLE
Add new solve time metrics

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** #119 

**This PR will result in the following new package version:** v0.12.1

We are adding new fields, but they should not impact existing customers and their data.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## 🚀 Feature Updates 🚀
- Added `solve_time_in_calendar_minutes` and `solve_time_in_business_minutes` to our `zendesk__ticket_metrics` model, which calculates calendar and business minutes for when the ticket was in the 'new', 'open', 'hold', or 'pending' status.

## 🔎 Under the Hood 🔎 
- Updates to the seed files and seed file configurations for the package integration tests to align with changes introduced by the PR on `dbt_zendesk_source` in applying the `dbt_utils.star` macro [PR #42](https://github.com/fivetran/dbt_zendesk_source/pull/42).

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [NA] dbt run (if incremental models are present)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:
(Sharing via Slack)

### CODE REVIEW

File: integration_tests/dbt_project.yml [hunk 1]
```diff
@@ -20,15 +20,18 @@ vars:
...
+    zendesk_group_identifier:                  "group_data" 
+    zendesk_organization_tag_identifier:       "organization_tag_data"
+    zendesk_user_identifier:                   "user_data"
+    zendesk_user_tag_identifier:               "user_tag_data"
 
 seeds:
-  +quote_columns: "{{ true if target.type == 'redshift' else false }}"
+
   zendesk_integration_tests:
+    +quote_columns: "{{ true if target.type == 'redshift' else false }}"
```
Comment (fivetran-jessicacherny) [inline integration_tests/dbt_project.yml:32]: why is there a condition on redshift?

File: models/agent_work_time/int_zendesk__ticket_work_time_calendar.sql [hunk 2]
```diff
@@ -12,6 +12,8 @@ with ticket_historical_status as (
             else 0 end as agent_wait_time_in_minutes,
         case when status in ('new', 'open', 'hold') then status_duration_calendar_minutes
             else 0 end as requester_wait_time_in_minutes,
+        case when status in ('new', 'open', 'hold', 'pending') then status_duration_calendar_minutes
```
Comment (fivetran-jessicacherny) [inline models/agent_work_time/int_zendesk__ticket_work_time_calendar.sql:15]: looks good!

File: models/agent_work_time/int_zendesk__ticket_work_time_business.sql [hunk 3]
```diff
@@ -117,6 +117,8 @@ with ticket_historical_status as (
           else 0 end as agent_wait_time_in_minutes,
       case when ticket_status in ('new', 'open', 'hold') then scheduled_minutes
           else 0 end as requester_wait_time_in_minutes,
+      case when ticket_status in ('new', 'open', 'hold', 'pending') then scheduled_minutes
+          else 0 end as solve_time_in_minutes,
```
Comment (fivetran-jessicacherny) [inline models/agent_work_time/int_zendesk__ticket_work_time_business.sql:121]: looks good!

File: models/agent_work_time/int_zendesk__ticket_work_time_calendar.sql [hunk 4]
```diff
@@ -12,6 +12,8 @@ with ticket_historical_status as (
             else 0 end as agent_wait_time_in_minutes,
         case when status in ('new', 'open', 'hold') then status_duration_calendar_minutes
             else 0 end as requester_wait_time_in_minutes,
+        case when status in ('new', 'open', 'hold', 'pending') then status_duration_calendar_minutes 
+            else 0 end as solve_time_in_minutes,
```
Comment (fivetran-jessicacherny) [inline models/agent_work_time/int_zendesk__ticket_work_time_calendar.sql:16]: can we go over why this is defined twice in two separate places? Is there an opportunity for us to follow DRY principles?

Comment (fivetran-avinash) [reply models/agent_work_time/int_zendesk__ticket_work_time_calendar.sql:16]: Hi @fivetran-jessicacherny ! These metrics are being constructed for two different models here--with schedules enabled in Zendesk, and without schedules. 

- The `int_zendesk__ticket_work_time_calendar.solve_time_in_minutes` field is when schedules are disabled (as I believe they are internally). This is the field being brought downstream into our internal models.
- The `int_zendesk__ticket_work_time_business.solve_time_in_minutes` field is when schedules are enabled (as they are for many other customers). This field is not currently being brought downstream internally.

Let me know if that clears it up!

File: integration_tests/dbt_project.yml (hunk 1 above)
Comment (fivetran-avinash) [reply integration_tests/dbt_project.yml:32]: It appears to be an artifact of our CircleCI integration. I'm not sure if we still need it but at the moment it isn't really blocking anything, so we'll discuss it internally. Thanks for spotting!

File: models/agent_work_time/int_zendesk__ticket_work_time_calendar.sql (hunk 4 above)
Comment (fivetran-jessicacherny) [reply models/agent_work_time/int_zendesk__ticket_work_time_calendar.sql:16]: definitely does! thank you @fivetran-avinash!

**Comment by fivetran-jessicacherny** [comment]
Test query I used that found some large differences between [hours_open](https://github.com/fivetran/analytics/blob/e84304774b56e52d355dd266cff168b7e7c551c7/dbt_ft_prod/models/bi_core/tickets.sql#L75) and `open_status_duration_in_calendar_minutes`

```
with tbl as (
  select 
    id,
    round(coalesce(hours_open, 0), 1) as hours_open,
    round(coalesce(solve_time_in_calendar_minutes, 0)/60, 1) as solve_time_in_calendar_hours
  from z_dbt_jcherny.tickets
)

select *,
  abs(hours_open - solve_time_in_calendar_hours) as absolute_diff
from tbl
where hours_open != solve_time_in_calendar_hours
  and id not like '5003%'
order by absolute_diff desc
```

File: package-lock.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  revision: feature/dbt-utils-star
+- package: calogica/dbt_date
+  version: 0.10.0
+- package: fivetran/fivetran_utils
+  version: 0.4.8
+- package: dbt-labs/spark_utils
+  version: 0.3.0
+- package: dbt-labs/dbt_utils
+  version: 1.1.1
+sha1_hash: 97acf24fa22aa67e32cd37ee9451f8e17d585973
```
Comment (fivetran-joemarkiewicz) [inline package-lock.yml:12]: This should be removed as we do not want this in the root directory of the package. This _could potentially_ interfere with customers projects.

File: packages.yml [hunk 6]
```diff
@@ -1,6 +1,10 @@
...
+  - git: https://github.com/fivetran/dbt_zendesk_source.git
+    revision: feature/dbt-utils-star
+    warn-unpinned: false 
+  # - package: fivetran/zendesk_source
+  #   version: [">=0.9.0", "<1.0.0"]
+
 
   - package: calogica/dbt_date
-    version: [">=0.9.0", "<1.0.0"]
+    version: [">=0.6.0", "<1.0.0"]
```
Comment (fivetran-joemarkiewicz) [inline packages.yml:10]: Reminder to switch before merge.

File: packages.yml [hunk 7]
```diff
@@ -1,6 +1,10 @@
 packages:
-  - package: fivetran/zendesk_source
-    version: [">=0.9.0", "<0.10.0"]
+  - git: https://github.com/fivetran/dbt_zendesk_source.git
+    revision: feature/dbt-utils-star
+    warn-unpinned: false 
+  # - package: fivetran/zendesk_source
+  #   version: [">=0.9.0", "<1.0.0"]
+
```
Comment (fivetran-joemarkiewicz) [inline packages.yml:7]: Reminder to switch before merge

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,11 @@
+# dbt_zendesk v0.12.1
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:1]: Given the possibility of the source being a breaking change, let's also make this a breaking change to reflect that update.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
These changes look good to me! I do have a comment regarding the package-lock.yml being removed and possibly making these breaking changes. Other than that, this is good to go!

File: integration_tests/requirements.txt [hunk 9]
```diff
@@ -4,6 +4,6 @@ dbt-redshift>=1.3.0,<2.0.0
 dbt-postgres>=1.3.0,<2.0.0
 dbt-spark>=1.3.0,<2.0.0
 dbt-spark[PyHive]>=1.3.0,<2.0.0
-dbt-databricks>=1.3.0,<2.0.0
+dbt-databricks>=1.6.0,<2.0.0
 
 oscrypto @ git+https://github.com/wbond/oscrypto.git@d5f3437
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply integration_tests/requirements.txt:9]: updated

File: package-lock.yml (hunk 5 above)
Comment (fivetran-avinash) [reply package-lock.yml:12]: Removed

File: integration_tests/package-lock.yml [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+  revision: feature/dbt-utils-star
+- package: calogica/dbt_date
+  version: 0.10.0
+- package: fivetran/fivetran_utils
+  version: 0.4.8
+- package: dbt-labs/spark_utils
+  version: 0.3.0
+- package: dbt-labs/dbt_utils
+  version: 1.1.1
+sha1_hash: 570155077da10b186bf3640e96c2b6746d0fc6dc
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply integration_tests/package-lock.yml:13]: removed

File: packages.yml (hunk 6 above)
Comment (fivetran-avinash) [reply packages.yml:10]: switched

**Pull request merged**